src/basic/socket-util.c

```diff
@@ -1560,7 +1560,6 @@ int socket_autobind(int fd, char **ret_name) {
          * "autobind" feature, but uses 64-bit random number internally. */
 
         assert(fd >= 0);
-        assert(ret_name);
```

Comment:
This one seems no longer necessary? I think we generally want the path here, so let's not be overzealous for now?

Reply:
this PR contains to places where we don't really care about the bound name.

```
$ git grep socket_autobind
src/basic/socket-util.c:int socket_autobind(int fd, char **ret_name) {
src/basic/socket-util.h:int socket_autobind(int fd, char **ret_name);
src/libsystemd/sd-netlink/test-netlink.c:        ASSERT_OK(socket_autobind(unix_fd, /* ret_name= */ NULL));
src/shared/notify-recv.c:        r = socket_autobind(fd, &path);
src/test/test-socket-netlink.c:        ASSERT_OK(socket_autobind(unix_fd, /* ret_name= */ NULL));
```